impls/src/backends/db.rs

```diff
@@ -146,6 +154,168 @@ impl Store {
 		Batch { store: self }
 	}
 
+	/// get an TxLogEntry by parent_key_id (key column) and tx_id, tx_slate_id
+	/// If no input is passed, returns all TxLogEntry transactions.
+	/// If outstanding_only = true then return Received/Sent not confirmed transactions.
+	pub fn get_txs(
+		&self,
+		tx_id: Option<u32>,
+		tx_slate_id: Option<Uuid>,
+		parent_key_id: Option<Vec<u8>>,
+		outstanding_only: bool,
+	) -> Vec<Serializable> {
+		// initial query (get all TxLogEntry)
+		let mut query = format!(
+			"SELECT * FROM data WHERE prefix = '{}' ",
+			TX_LOG_ENTRY_PREFIX
+		);
+
+		// filter by parent_key_id (key)
+		query = match parent_key_id {
+			Some(key) => format!(
+				"{} {} '{}'",
+				query,
+				SQLITE_FILTER,
+				String::from_utf8(key).unwrap()
+			),
+			None => query,
+		};
```

Comment:
I believe that the parent_key_id is different from the key.
Also keep it as the original type `Identifier` and use serd_json to convert it to string.